db/queries.go

```diff
@@ -67,33 +75,39 @@ func (handler Handler) GetAppsForCollection(collectionID string) ([]RebbleApplic
 	}
 	json.Unmarshal(appIdsB, &appIds)
 
+	for i, appId := range appIds {
+		appIds[i] = "'" + appId + "'"
+	}
+
+	idList := strings.Join(appIds, ", ")
+
 	apps := make([]RebbleApplication, 0)
-	for _, id := range appIds {
-		rows, err = handler.Query("SELECT id, name, type, thumbs_up, screenshots, published_date, supported_platforms FROM apps WHERE id=?", id)
+
+	// It is not possible to give a list or an order via a prepared statement, so this will have to do. Unless someone managed to change their application ID to a SQL injection, this is perfectly safe.
+	rows, err = handler.Query("SELECT id, name, type, thumbs_up, screenshots, published_date, supported_platforms FROM apps WHERE id IN (" + idList + ") ORDER BY " + order + " DESC")
+	if err != nil {
+		return nil, err
+	}
+
+	for rows.Next() {
+		app := RebbleApplication{}
+		var t int64
+		var supported_platforms_b []byte
+		var screenshots_b []byte
+		err = rows.Scan(&app.Id, &app.Name, &app.Type, &app.ThumbsUp, &screenshots_b, &t, &supported_platforms_b)
 		if err != nil {
-			return nil, err
+			return []RebbleApplication{}, err
```

Comment:
Since we are potentially returning before finishing our rows.Next() loop, should we defer rows.Close() after we open the Query? I think we would potentially leave unopened connections otherwise.